src/bio/bio_wal.c

```diff
@@ -1758,6 +1761,12 @@ bio_wal_replay(struct bio_meta_context *mc, struct bio_wal_rp_stats *wrs,
 			tight_loop = 0;
 			bio_yield(NULL);
 		}
+
+		/* test need generate enough tx */
+		if (nr_replayed > 1000 && DAOS_FAIL_CHECK(DAOS_WAL_FAIL_REPLAY)) {
```

Comment:
Minor: When the caller set FAIL_LOC, it could also set the FAIL_VALUE, so that we can compare the 'nr_replayed' with the FAIL_VALUE set by caller instead of hardcoded 1000.